src/AccessibilityInsights.SharedUx/Telemetry/TelemetrySink.cs

```diff
@@ -0,0 +1,94 @@
+    /// It should not be accessed directly, but only through the higher-level telemetry classes
+    /// such as Logger and TelemetryController.
+    /// This class should not throw any exceptions.
+    /// </summary>
+    internal static class TelemetrySink
+    {
+        private static ITelemetry _telemetry => Container.GetDefaultInstance()?.Telemetry;
+
+        /// <summary>
+        /// Whether or not telemetry toggle button is enabled in the settings.
+        /// </summary>
+        public static bool IsTelemetryAllowed { get; internal set; } = false;
+
+        /// <summary>
+        /// Whether or not telemetry is enabled. Exposed to allow callers who do lots of
+        /// work to short-circuit their processing when telemetry is disabled
+        /// </summary>
+        public static bool IsEnabled => IsTelemetryAllowed && _telemetry != null;
+
+        /// <summary>
+        /// Publishes event with single property/value pair to the current telemetry pipeline
+        /// </summary>
+        /// <param name="action"></param>
+        /// <param name="property"></param>
+        /// <param name="value"></param>
+        public static void PublishTelemetryEvent(string action, string property, string value)
+        {
+            PublishTelemetryEvent(action, new Dictionary<string, string>
+            {
+                        { property, value }
+                    });
+        }
+
+        /// <summary>
+        /// Publishes event to the current telemetry pipeline
+        /// </summary>
+        /// <param name="action">The action being recorded</param>
+        /// <param name="propertyBag">Associated property bag--this may be null</param>
+        public static void PublishTelemetryEvent(string action, IReadOnlyDictionary<string, string> propertyBag = null)
+        {
+            if (!IsEnabled) return;
+
+            try
+            {
+                _telemetry.PublishEvent(action, propertyBag);
+            }
+            catch (Exception) { }
+        }
+
+        /// <summary>
+        /// Explicitly updates context properties to be appended to future calls to the current telemetry pipeline
+        /// </summary>
+        /// <param name="property"></param>
+        /// <param name="value"></param>
+        public static void AddOrUpdateContextProperty(string property, string value)
+        {
+            if (!IsEnabled) return;
+
+            try
+            {
+                _telemetry.AddOrUpdateContextProperty(property, value);
+            }
+            catch {}
+        }
+
+        /// <summary>
+        /// Report an Exception into the pipeline
+        /// </summary>
+        /// <param name="e">The Exception to report</param>
+        public static void ReportException(Exception e)
+        {
+            if (!IsEnabled) return;
+            if (e == null) return;
+
+            try
+            {
+                _telemetry.ReportException(e);
+            }
+            catch {}
```

Comment:
For consistency with line 59,
```suggestion
            catch (Exception) { }
```